### PR TITLE
H2 2.0 support

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-15014
https://hibernate.atlassian.net/browse/HHH-15009
https://hibernate.atlassian.net/browse/HHH-15035

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java [hunk 1]
```diff
@@ -2379,12 +2384,33 @@ public static void format_toVarchar(QueryEngine queryEngine) {
...
 						queryEngine.getTypeConfiguration().getBasicTypeRegistry().resolve( StandardBasicTypes.STRING )
 				)
-				.setExactArgumentCount( 2 )
-				.setParameterTypes(TEMPORAL, STRING)
+				.setArgumentsValidator( formatValidator() )
 				.setArgumentListSignature( "(TEMPORAL datetime as STRING pattern)" )
 				.register();
 	}
 
+	public static ArgumentsValidator formatValidator() {
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java:2392]: This is a good idea, but have fun with the conflicts.

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java:2392]: I fixed them already :)

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java:2392]: Oh, I see, no conflicts.

**Review by gavinking — COMMENTED** [review]
Wait, @beikov, I thought this validation was a good idea, but then I realized that it actually duplicates what already happens in `SemanticQueryBuilder.visitFormat()`. It _is_ a good idea, and that's why I already had it. :-)

Now, doing that validation in an `ArgumentsValidator` is, I dunno, _not obviously better_? SQB already has the job of interpreting/validating literal values, so I don't see this validation as being out of place where it is right now.

**Pull request merged**